### PR TITLE
Check config.json before applying system defaults in gen_config

### PR DESCRIPTION
**Pull request by rickzx**
In some config.json, it already provides chat-related config, like bos_token_id, eos_token_id, temperature, etc. We should check the provided config.json before applying the system defaults.

### CODE REVIEW

**Comment by junrushao** [comment]
Interesting. Previously I was assuming generation-related configurations are usually under `generation_config.json`. Would you mind sharing a specific case where `bos_token_id` or `temperature` exist in `config.json` rather than `generation_config.json`?

**Comment by junrushao** [comment]
Ah got it, thanks for the clarification! To organize the logic a bit, let's instead update "Step 2" which loads `generation_config.json` as follows:

```python
for generation_config_filename in ["generation_config.json", "config.json"]:
  ...
```

**Comment by rickzx** [comment]
@junrushao Good point! I updated the commit and here's an example output
```
[2023-12-07 16:50:21] INFO gen_config.py:109: Not found generation_config.json: /ssd1/rickzhou/rick_fork/mlc-llm/models/music-small-800k/generation_config.json
[2023-12-07 16:50:21] INFO gen_config.py:101: Found config.json: /ssd1/rickzhou/rick_fork/mlc-llm/models/music-small-800k/config.json
[2023-12-07 16:50:21] INFO gen_config.py:107: [config.json] Setting bos_token_id: 55025
[2023-12-07 16:50:21] INFO gen_config.py:107: [config.json] Setting eos_token_id: 55025
[2023-12-07 16:50:21] INFO gen_config.py:107: [config.json] Setting temperature: 1.0
[2023-12-07 16:50:21] INFO gen_config.py:107: [config.json] Setting top_p: 1.0
[2023-12-07 16:50:21] INFO gen_config.py:107: [config.json] Setting repetition_penalty: 1.0
[2023-12-07 16:50:21] INFO gen_config.py:107: [config.json] Setting pad_token_id: None
```

File: python/mlc_chat/compiler/gen_config.py [hunk 1]
```diff
@@ -94,18 +94,19 @@ def gen_config(  # pylint: disable=too-many-locals,too-many-arguments,too-many-b
...
-            if hasattr(mlc_chat_config, key) and getattr(mlc_chat_config, key) is None:
-                setattr(mlc_chat_config, key, value)
-                logger.info("[generation_config.json] Setting %s: %s", bold(key), value)
-    else:
-        logger.info("%s generation_config.json: %s", NOT_FOUND, generation_config)
+    # Step 2. Load `generation_config.json` and `config.json` for text-generation related configs
+    for generation_config_filename in ["generation_config.json", "config.json"]:
+        generation_config = config.parent / generation_config_filename
+        if generation_config.exists():
+            logger.info("%s %s: %s", FOUND, generation_config_filename, generation_config)
```
Comment (junrushao) [inline python/mlc_chat/compiler/gen_config.py:101]: probably we don't need this line

**Review by junrushao — APPROVED** [review]
Looks nice! Let's get it in - and just a small nit

File: python/mlc_chat/compiler/gen_config.py (hunk 1 above)
Comment (rickzx) [reply python/mlc_chat/compiler/gen_config.py]: Removed

**Pull request merged**